### ISSUE TITLE
Include types in package

### ISSUE THREAD

**Issue by jamiehaywood** [mifi/react-lottie-player#5]
Really like the package, would be great to include the types for TypeScript support. If I find the time I will try and raise a PR to do this.

**Comment by mifi** [comment]
Sure, PR welcome!

**Comment by richt3211** [comment]
Wanted to +1 this. @jamiehaywood any update?

**Comment by aleksmizin** [comment]
+1

**Comment by mikejdegroot** [comment]
+1

**Comment by skoruba** [comment]
+1

**Comment by skoruba** [comment]
I've use this one and it works:

The properties `rendererSettings` and `audioFactory` should be defined probable better, I am not sure how, mabye someone help with the improvements. 😊

```
declare module 'react-lottie-player' {
  import React from 'react';

  interface LottieProps {
    animationData?: any;
    path?: string;
    play?: boolean;
    goTo?: number;
    speed?: number;
    direction?: number;
    loop?: number | boolean;
    segments?: number[] | boolean;
    rendererSettings?: any;
    renderer?: string;
    audioFactory?: any;
    onComplete?: () => void;
    onLoopComplete?: () => void;
    onEnterFrame?: () => void;
    onSegmentStart?: () => void;
    style?: React.CSSProperties;
  }

  export default class Lottie extends React.Component<LottieProps> {}
}

```

**Comment by GregdTd** [comment]
Any update here ?

**Comment by skoruba** [comment]
I added a PR, but I do not know how microbundle works, can you check my PR? Thx

**Comment by jamiehaywood** [comment]
Apologies, I moved job a couple of weeks after raising this issue, and forgot to reply 🤦🏻‍♂️.
A couple of things regarding the PRs:

1. I've taken a look into microbundle-crl and it doesn't look like it's got anything for emitting / including types in the build, so what you'll probably have to do is add another step to the build process along the lines of `cp index.d.ts /dist`
2. I would recommend naming the types file `index.d.ts` (whichever types PR is merged) - it's a pretty standard naming convention

**Comment by mifi** [comment]
this is now fixed in 1.3.2. please test and reopen if any issues with the types
